test/e2e/automated_cleaning_test.go

```diff
@@ -0,0 +1,240 @@
+
+		By("Waiting for the BMH to become provisioned")
+		WaitForBmhInProvisioningState(ctx, WaitForBmhInProvisioningStateInput{
+			Client: clusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  metal3api.StateProvisioned,
+		}, e2eConfig.GetIntervals(specName, "wait-provisioned")...)
+
+		By("Deprovisioning the BMH to trigger automated cleaning")
+		helper, err := patch.NewHelper(&bmh, clusterProxy.GetClient())
+		Expect(err).NotTo(HaveOccurred())
+		bmh.Spec.Image = nil
+		bmh.Spec.UserData = nil
+		Expect(helper.Patch(ctx, &bmh)).To(Succeed())
+
+		By("Waiting for the BMH to be in deprovisioning state")
+		WaitForBmhInProvisioningState(ctx, WaitForBmhInProvisioningStateInput{
+			Client: clusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  metal3api.StateDeprovisioning,
+		}, e2eConfig.GetIntervals(specName, "wait-deprovisioning")...)
+
+		By("Waiting for the BMH to become available again")
+		WaitForBmhInProvisioningState(ctx, WaitForBmhInProvisioningStateInput{
+			Client: clusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  metal3api.StateAvailable,
+		}, e2eConfig.GetIntervals(specName, "wait-available")...)
+
+		By("Patching the BMH again to trigger re-provisioning")
+		userDataSecretName := "user-data-ssh-setup"
+		sshPubKeyPath := e2eConfig.GetVariable("SSH_PUB_KEY")
+		// Create new userdata secret for only SSH setup
+		createSSHSetupUserdata(ctx, clusterProxy.GetClient(), namespace.Name, userDataSecretName, sshPubKeyPath)
+		userDataSecret = &corev1.SecretReference{
+			Name:      userDataSecretName,
+			Namespace: namespace.Name,
+		}
+		err = PatchBMHForProvisioning(ctx, PatchBMHForProvisioningInput{
+			client:         clusterProxy.GetClient(),
+			bmh:            &bmh,
+			e2eConfig:      e2eConfig,
+			namespace:      namespace.Name,
+			userDataSecret: userDataSecret,
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for the BMH to be in provisioning state")
+		WaitForBmhInProvisioningState(ctx, WaitForBmhInProvisioningStateInput{
+			Client: clusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  metal3api.StateProvisioning,
+		}, e2eConfig.GetIntervals(specName, "wait-provisioning")...)
+
+		By("Waiting for the BMH to become provisioned")
+		WaitForBmhInProvisioningState(ctx, WaitForBmhInProvisioningStateInput{
+			Client: clusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  metal3api.StateProvisioned,
+		}, e2eConfig.GetIntervals(specName, "wait-provisioned")...)
+
+		By("Connecting via SSH to check disk state after cleaning")
+		client := EstablishSSHConnection(e2eConfig, bmc.IPAddress)
+		defer client.Close()
+
+		By("Checking that the first disk has been cleaned")
+		output, err := executeSSHCommand(client, "ls -la /test_file_vda.txt 2>/dev/null || echo 'file not found'")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(output).To(ContainSubstring("file not found"), "Test file /test_file_vda.txt should have been cleaned")
+
+		By("Verifying second disk exists but filesystem is cleaned")
```

Comment:
We would need to check this file before deprovisioning also, otherwise we cannot really trust that it was there to begin with.
Side note: If we could make this test work both ways it would be really awesome! I mean that we would test with cleaning enabled and see that the files disappear, and with cleaning disabled and see that they stay. It is probably better to leave that for another PR though 🙂